contracts/stabilizer/PegStabilityModule.sol

```diff
@@ -0,0 +1,276 @@
+        uint256 _oldMintFee = mintFeeBasisPoints;
+        mintFeeBasisPoints = newMintFeeBasisPoints;
+
+        emit MintFeeChanged(_oldMintFee, newMintFeeBasisPoints);
+    }
+
+    /// @notice internal helper function to set the redemption fee
+    function _setRedeemFee(uint256 newRedeemFeeBasisPoints) internal {
+        require(newRedeemFeeBasisPoints <= Constants.BASIS_POINTS_GRANULARITY, "PegStabilityModule: Redeem fee exceeds bp granularity");
+        uint256 _oldRedeemFee = redeemFeeBasisPoints;
+        redeemFeeBasisPoints = newRedeemFeeBasisPoints;
+
+        emit RedeemFeeChanged(_oldRedeemFee, newRedeemFeeBasisPoints);
+    }
+
+    /// @notice helper function to set reserves threshold
+    function _setReservesThreshold(uint256 newReservesThreshold) internal {
+        require(newReservesThreshold > 0, "PegStabilityModule: Invalid new reserves threshold");
+        uint256 oldReservesThreshold = reservesThreshold;
+        reservesThreshold = newReservesThreshold;
+
+        emit ReservesThresholdChanged(oldReservesThreshold, newReservesThreshold);
+    }
+
+    /// @notice helper function to set the target
+    function _setTarget(IPCVDeposit newTarget) internal {
+        require(address(newTarget) != address(0), "PegStabilityModule: Invalid new target");
+        IPCVDeposit oldTarget = target;
+
+        emit TargetChanged(oldTarget, newTarget);
+    }
+
+    /// @notice function to redeem FEI for an underlying asset
+    function redeem(address to, uint256 amountFeiIn) external virtual override nonReentrant whenNotPaused returns (uint256 amountOut) {
+        updateOracle();
+        Decimal.D256 memory price;
+
+        (amountOut, price) = _getRedeemAmountOutAndPrice(amountFeiIn);
+        require(_validPrice(price), "PegStabilityModule: price out of bounds");
+        FEI.transferFrom(msg.sender, address(this), amountFeiIn);
+        FEI.burn(amountFeiIn);
+
+        token.safeTransfer(to, amountOut);
+
+        emit Redeem(to, amountFeiIn);
+    }
+
+    /// @notice function to buy FEI for an underlying asset
+    function mint(address to, uint256 amountIn) external virtual override payable nonReentrant whenNotPaused returns (uint256 amountFeiOut) {
+        require(msg.value == 0, "PegStabilityModule: cannot send eth to mint");
+
+        updateOracle();
+        Decimal.D256 memory price;
+
+        (amountFeiOut, price) = _getMintAmountOutAndPrice(amountIn);
+        require(_validPrice(price), "PegStabilityModule: price out of bounds");
+
+        token.safeTransferFrom(msg.sender, address(this), amountIn);
+
+        _mintFei(msg.sender, amountFeiOut);
+
+        emit Mint(to, amountIn);
+    }
+
+    /// @notice function to pause the PSM. Only accessible in the ERC20PSM
+    /// should be called whenever the DAI price is above $1.02 or below $0.98 as this will pause the contract
+    /// then a guardian or governor will have to unpause the contract
+    function oracleErrorPause() external virtual whenNotPaused {
```

Comment:
What exactly does this method do? The _validPrice logic already short circuits redemptions and mints